### DISCUSSION TITLE
Problem 'python3: command not found' using docker/mamba

### DISCUSSION THREAD

**Discussion by sleyn** [nextflow-io/nextflow#3070] [Q&A]
Hello!

I'm trying to a run script in a docker container.

The docker container is based on the micromamba environment.

I can use the container with the script from my terminal and everything works fine.
However when I'm trying to use the same script from the Nextflow I'm getting an error:

```
Command exit status:
  127

Command output:
  (empty)

Command error:
  .command.sh: line 2: python3: command not found
```

The docker file
```
FROM mambaorg/micromamba:0.24.0
COPY --chown=$MAMBA_USER:$MAMBA_USER ./scripts/environment.yml /apps/environment.yml

RUN micromamba install -y -n base -f /apps/environment.yml && \
    micromamba clean --all --yes

ARG MAMBA_DOCKERFILE_ACTIVATE=1

WORKDIR /apps/
COPY scripts/patric2fasta.py /apps
ENV PATH=/apps:${PATH}
```

I was trying to use a shebang `#!/usr/bin/env python3` to the script.

An error just changed to
```
Command error:
  /usr/bin/env: ‘python3’: No such file or directory
```

The Nextflow process is:
```
process TBL_TO_FASTA {
    container 'semenleyn/patric2fasta:latest'
    containerOptions '-v "$(pwd):/temp"'
    publishDir "${launchDir}/output/family_fasta/", mode: 'copy'

    input:
        tuple val(gene), val(pgfam), val(atcc_sequence)
        path family_table

    output:
        path "${pgfam}.fasta", emit: family_fasta_ch

    script:
    """
    python3 patric2fasta.py -t /temp/${family_table} -o ${pgfam}.fasta --ref_sequence ${atcc_sequence}
    """
}
```

It is not the first time I'm using docker containers but the first time I'm using it with mamba. Does anyone have an idea what is wrong?

Thanks,
Semion

**Comment by sleyn** [comment]
UPD: I was made it work using absolute paths inside the container:
```
    """
    /opt/conda/bin/python3 /apps/patric2fasta.py -t /temp/${family_table} -o ${pgfam}.fasta --ref_sequence ${atcc_sequence}
    """
```

However I still don't understand why I can use simple `python3 patric2fasta.py` in Terminal and it fails in Nextflow.

**Reply by pditommaso** [reply]
TLDR; current version of nextflow overrides the container entrypoint. The upcoming release should fix it 

https://nextflow.slack.com/archives/C02T98A23U7/p1658907452181369?thread_ts=1658871047.388969&cid=C02T98A23U7

**Reply by sleyn** [reply]
Thank you!
